climada/hazard/trop_cyclone.py

```diff
@@ -151,7 +151,17 @@ def set_from_tracks(self, tracks, centroids=None, description='',
             coastal_idx = ((centroids.dist_coast < INLAND_MAX_DIST_KM * 1000)
                            & (np.abs(centroids.lat) < 61)).nonzero()[0]
 
-        LOGGER.info('Mapping %s tracks to %s centroids.', str(tracks.size),
+        # Restrict to coastal centroids within reach of any of the tracks
+        t_bounds = tracks.get_bounds(deg_buffer=CENTR_NODE_MAX_DIST_DEG)
+        t_mid_lon = 0.5 * (t_bounds[0] + t_bounds[2])
+        coastal_centroids = centroids.coord[coastal_idx]
+        co.lon_normalize(coastal_centroids[:, 1], center=t_mid_lon)
+        coastal_idx = coastal_idx[((t_bounds[0] <= coastal_centroids[:, 1])
+                                   & (coastal_centroids[:, 1] <= t_bounds[2])
+                                   & (t_bounds[1] <= coastal_centroids[:, 0])
+                                   & (coastal_centroids[:, 0] <= t_bounds[3]))]
```

Comment:
I did some performance testing and found these lines to introduce no significant slow-down in cases where it is redundant, e. g. if there is only a single track or in cases where the centroids are already chosen tightly around the track set.

Reply:
Could you expand the t_bounds to avoid the hard-coded indexing which is always very obscure to the reader? I actually do not know how to interpret them, hence my example below is probably incorrect.

```
[tr_lat_min, tr_lat_max, tr_lon_min, tr_lon_max] = tracks.get_bounds(deg_buffer=CENTR_NODE_MAX_DIST_DEG)
```